some changes for work with params

MStruller opened the pull request:
most of changes have effect in view translate layout edit

in view translate layout edit:
    we can now work with JForm, used for Parameters and at this moment for field 'category' (in contentelemtnt content)

in view translate layout default:
    add org language filter
    for native translation i have change the function createContentSQL() and countContentSQL() in ContentElement.php

```
    to get all items also for different languages
        not only selected language and * 

        an example:

        we have two user
        1. User : write in en-GB
        2. User : write in de-DE

        both Users create an article 1.User over monkeys in en-GB
         the 2.User over flowers in de-DE and also create an article over buildings in *
        and after create 
        what we want to see in the overview

        1. User want to translate the flowers. See the user it whitout the changes belowabove?

        extend the #__jf_translationmap with field org_reference_id ?
        or extend the #__jf_translationmap with field time-stamp?
        example:
        we have an 3. User: write in it-IT
        what article will display here in overview? 
        both english and german?
        only german?
        only english?

        and which is the base article to translate?

        what is if the users have create and translate the articles without joomfish
        make an extra view for manuel set translationmap? (see under)

also add an new view translationmap (startpoint is in view translate layout default only have is an field 'language') to set translationmap manual 
if we have existing j1.6/2.5 translation not make with joomfish
```

contentelement categories:
    add field extension and filter extension and also in xml '<filter>c.id > 1 </filter>' to hide the root

Fix for editTranslation if new and apply

other way for 3rd party where want use contentelements and TranslationObject:
    for this i have add in classes/JoomfishManager.class.php
    includePaths can set over JoomfishManager or request

```
have add functions to load contentelements over request.
and can load other path over the xml (contentelement )
    <treatment> 
        is not needed if use in ContentElement.php getTranslationObjectClass
        <translationObject>TranslationObjectContent</translationObject>
        is not needed if use in ContentElement.php getTranslateFormsClass
        <translateForms>TranslateFormsContent</translateForms>

        see also comment in ContentElement.php

        for other paths we can add in the treatment

        <includePath>relative to JPATH_ROOT</includePath>
        or 
        <includePath extension="1" site="0">jf</includePath>
        <extension>com_content</extension>
    <treatment>
ContentElement,TranslationFilter,TranslateForms,TranslateModel,TranslationObject and TranslateParams
can load over the treatment->includePath
(see in JoomfishManager function _loadContentElements(), ContentElement function getTranslationObjectClass(), TranslationFilter function getTranslationFilters() )

so developer of other components there use joomfish must not copy all the needed files
or can create some on the fly if needed.
```

others:

folder contentelements moved to models/jf/xmls

all Filters (also TranslationFilter.php) is moved from contentelements and models to models/jf/filters
all Filters in seperate files

all TranslateParams is moved from models to models/jf/params
all TranslateParams in seperate files

created models/jf/forms
and created TranslateForms all in seperate files

created models/jf/models
and created TranslateModel all in seperate files

created models/jf/models
and created TranslateModel all in seperate files

all Objects is moved from models to models/jf/objects
all TranslationObject in seperate files

moved ContentElement.php, ContentElementTable.php and ContentElementTableField.php to models/jf/element

all above is do for easy find the classes and files
for work with 'params' 
and also other special fields like 'category' (use JForm for output like params)

make category field work (over new TranslateForms)

The params and field category work and save
but only Content, Menus and Modules at this moment work with the new classes (TranslateParams, TranslateForms) 
for the other components the work must done.

Fix for editTranslation if new and apply

add org language filter and fix countContentSQL and createContentSQL (only for native)

add an manual translationmap view to set translationmap manual if we have existing j1.6/2.5 translation not make with joomfish

also change some double space with tab

MStruller committed:
prepare JForm control

MStruller committed:
prepare JForm control

MStruller committed:
orig and trans params in one slider panel

MStruller committed:
Missing (un)published icons at the bottom of Translate/Orphans views and Control Panels Unpublished Content Elements tab. - II

MStruller committed:
contenelement filename to tablename

MStruller committed:
some changes for translate params

MStruller committed:
folder contentelements moved to models/jf/xmls

all Filters (also TranslationFilter.php) is moved from contentelements and models to models/jf/filters
all Filters in seperate files

all TranslateParams is moved from models to models/jf/params
all TranslateParams in seperate files

created models/jf/forms
and created TranslateForms all in seperate files

created models/jf/models
and created TranslateModel all in seperate files

created models/jf/models
and created TranslateModel all in seperate files

all Objects is moved from models to models/jf/objects
all TranslationObject in seperate files

moved ContentElement.php, ContentElementTable.php and ContentElementTableField.php to models/jf/element

all above is do for easy find the classes and files
for work with 'params'
and also other special fields like 'category' (use JForm for output like params)

make category field work (over new TranslateForms)

The params and field category work and save
but only Content, Menus and Modules at this moment work with the new classes (TranslateParams, TranslateForms)
for the other components the work must done.

Fix for editTranslation if new and apply

add org language filter and fix countContentSQL and createContentSQL (only for native)

MStruller committed:
Delete Thumbs.db

MStruller committed:
Add views/translate/tmpl/translationmap.php
to set translationmap manual.
Change to load contentElement, TranslationFilter...

akempkens commented:
Why did you moved the content elements to the models folder? Due to compatibility reasons I'm not really happy with this move. 

For 3rd party developers we have the idea to allow them to use their own directory and not JoomFish at all. Therefore we created the logic to use our interface for translation objects.

May be you can explain your change a bit more.

Alex

MStruller commented:
As first the location for the contentelements is for my changes not necessary.

An question: 'compatbility reasons' compattible with what? joomfish for j1.5?

i have moved the content elements to get an same structure for the other i have create ore move to the models/jf folder
this was an result by the work on the params and jform
I have add an function getExtraPath() in helpers/extensionHelper.php so we can change easy the path
all functions where need the contenelemts use the getExtraPath() so only in helper we need to change back to com_joomfish/contentelements.

For use contentelements in other paths (3rd party) i have also add an routine to get the path
see JoomfishManager.class.php function addIncludePath and controllers/translate.php __construct().

for 3rd party i have make functions to get all needed files from own path and/or from the joomfish path
see in models/jf/element/ContentElement.php getTranslateParamsClass(),getTranslationObjectClass(), getTranslateFormsClass()dificult

i know there are some changes and i have try to descript it (see at the head)
also i know the commits are very unsortet.

To explain all the changes is difficult (not only in english)
so let me know where you have problems to understand so i can respond it.

Micha

MStruller committed:
change .gitignore to original

MStruller committed:
change manifest.xml to original

MStruller committed:
layout translationmap
use in layout default default_list
to set the table #__jf_translationmap manual
if the translation not make with joomfish

MStruller committed:
remove newline?

MStruller commented on component/admin/classes/JoomfishManager.class.php:
```diff
@@ -81,9 +89,108 @@ class JoomFishManager {
+	 *
+	 * @return	array	An array with directory elements
+	 */
+
```
for use from 3rd party also use in controllers/translate.php in __construct to get the includePath over request
(i have make for 'pagesanditems' an extension 'pi_fish' to translate the pagesanditems fields and i use joomfish in an iframe to get the joomfish edit translate view, also i use derivates from the contentelement 'content.xml' and with an includePath i can store the contentelements on  an place i want9

MStruller commented on component/admin/classes/JoomfishManager.class.php:
```diff
@@ -188,14 +327,63 @@ private function _loadContentElements() {
-				if ($xmlDoc->load(JOOMFISH_ADMINPATH . "/contentelements/" . $file)) {
+				
+				//if ($xmlDoc->load(JOOMFISH_ADMINPATH . "/contentelements/" . $file)) {
+				if ($xmlDoc->load(JoomfishExtensionHelper::getExtraPath('xmls').DS. $file)) {
```
here we load the file with the path is in the JoomfishExtensionHelper
in the JoomfishExtensionHelper we can set the path back from models/jf/xmls to contentelements

MStruller commented on component/admin/classes/JoomfishManager.class.php:
```diff
@@ -188,14 +327,63 @@ private function _loadContentElements() {
+							$contentElement = null;
+							$tableName = self::getTableName($xmlDoc);
+							$treatment = self::getTreatment($xmlDoc,$tableName);
+							if(count($treatment) > 0)
```
have thecxontentelement(xml) some instruction for us?
like an classname an extra path for the class
so we try to load this.

The same is also in function _loadContentElement()

MStruller commented on component/admin/controllers/translate.php:
```diff
@@ -111,6 +125,151 @@ function __construct()
 
 	}
 
```
for manual set the jf_translationmap table if we have translations not make with joomfish only treatment native
the function translationmap is call in view translation default list as an Squeezebox and the functiontranslationMapSave is call from the translationmap view

MStruller commented on component/admin/models/fields/htmltext.php:
```diff
@@ -0,0 +1,90 @@
+			return parent::getInput();
+		}
+	}
+}
```
this file is add to demonstrate what we can do with jform for view translate edit, but at this moment not used

MStruller commented:
are the comments above helpfull?
if so i will write more
i can also comment files where comments in the file
Micha

MStruller committed:
in J2.5 beta 2 and above we must add the table path
to com_menus/tables

MStruller committed:
add setFieldAttribute('catid','extension','com_content') for J2.5

akempkens commented:
Michael,

Please do NOT add additional changes to one single pull request. It makes it extremely hard for us to follow thru the changes and accept them. Meanwhile this pull request is out of synch with our repository and github can't merge it anymore.

I need to close this request and ask you to follow the steps as suggested by github.
1. create a fork from the latest version of the repository
2. create one pull request for one specific topic - and do not mix topics
3. if one request is applied ensure that for all other pull requests a merge with the latest version of the repository is in place
